dspace-api/src/main/java/org/dspace/authorize/AuthorizeServiceImpl.java

```diff
@@ -787,4 +802,186 @@ public ResourcePolicy createOrModifyPolicy(ResourcePolicy policy, Context contex
         return resourcePolicyService.findExceptRpType(c, o, actionID, rpType);
     }
 
+    /**
+     * Checks that the context's current user is a community admin in the site by querying the solr database.
+     *
+     * @param context   context with the current user
+     * @return          true if the current user is a community admin in the site
+     *                  false when this is not the case, or an exception occurred
+     */
+    @Override
+    public boolean isCommunityAdminSolr(Context context) throws SQLException {
+        return performCheck(context, "search.resourcetype:Community");
+    }
+
+    /**
+     * Checks that the context's current user is a collection admin in the site by querying the solr database.
+     *
+     * @param context   context with the current user
+     * @return          true if the current user is a collection admin in the site
+     *                  false when this is not the case, or an exception occurred
+     */
+    @Override
+    public boolean isCollectionAdminSolr(Context context) throws SQLException {
```

Comment:
as for isCommunityAdminSolr

Reply:
Agreed here as well...it's the same issue.  It'd be nice if the Solr code was moved directly into `isCollectionAdmin()` and this method was removed.  Again, though, if there's a reason for two implementations of `isCollectionAdmin` (one for DB and one for Solr), I'd be OK with moving this to a followup ticket